### PR TITLE
updated system tests code to update strimzi version to 0.40.0

### LINKED ISSUE #1040: Upgrade system tests to use Strimzi 0.40
To allow the system tests to use Kafka 3.7.0, we need to take a Strimzi 0.40, once it is ready.

### PR DESCRIPTION
**Pull request by franvila**
### Type of change

- Enhancement / new feature

### Description

Closes #1040. Need to update strimzi version to 0.40.0 just released. 

### Checklist

- [ ] Write tests
- [X] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Comment by franvila** [comment]
@strimzi-ci run system tests

File: kroxylicious-systemtests/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
     </description>
 
     <properties>
-        <kafka.version>3.6.1</kafka.version>
+        <kafka.version>3.7.0</kafka.version>
```
Comment (k-wall) [inline kroxylicious-systemtests/pom.xml:27]: Let's remove this override altogether.

**Review by k-wall — APPROVED** [review]
lgtm

**Comment by franvila** [comment]
@strimzi-ci run system tests

**Pull request merged**